Cassandra 4.0

dhirsch1 opened the pull request.

dhirsch1 committed:
First commit for the new version

dhirsch1 committed:
Fixed the nav

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Interim commit

dhirsch1 committed:
Commit of added flow diagrams

dhirsch1 committed:
Some word fixes to the Examples topic

dhirsch1 committed:
First commit of Reference updates

dhirsch1 committed:
Merge branch 'latest' into DOCS-9178-Cassandra-4.0-DH

dhirsch1 committed:
Entered word tweaks for the Reference

dhirsch1 committed:
Interim commit

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc:
```diff
@@ -0,0 +1,127 @@
++
+You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
++
+. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
```
Can you please add here a description of the security tab as well?  I think I've put it only on design center doc. Thanks

dhirsch1 committed:
Commit of small fixes

dhirsch1 committed:
Added a new Upgrade topic

kevintroller committed:
Removed CASSANDRA_EXCEPTION error type

valean27 commented on salesforce/10.5/modules/ROOT/pages/index.adoc:
```diff
@@ -35,7 +35,13 @@ If you plan to create the XML for your Mule app, include the correct namespace a
 
 * New users
 +
+<<<<<<< HEAD
```
>>>> These are some merging conflicts, please remove the wrong ones.

kevintroller committed:
Remove common use cases line

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+==== Add a Parse Template Component
+
+Add a *Parse Template* component to create a template for an HTML form. The example uses this form to convey information for creating the table. (*Reviewers, is this text correct?*):
```
yes

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+. From the Mule Palette view, search for `parse` and drag *Core* > *Parse Template* to the right of *Listener*.
+. In the *Location* field, enter `form.html`.
+*Reviewers, where is the information for this template? I assume it is in form.html, but where is the form put?*
```
form.html is located in main/resources of the demo

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+==== Add a Transform Message Component
+
+Add a *Transform Message* component to set Java variables based on the payload received from *HTTP Listener*: *Reviewers, is this description correct*?
```
yes

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+==== Add the Create Keyspace Operation
+
+Add the *Create keyspace* operation to create the new keyspace for the new table: (*Reviewers: please correct and make this text better*)
```
Create a keyspace for the new table to exist in. The tables are created in a keyspace, so you need to create a keyspace first.

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+=== Add a Transform Message Component
+
+Add a *Transform Component* to set Java variables based on the payload received from *HTTP Listener*: ()*Reviewers, is this description correct*)?
```
yes

valean27 commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+==== Add the Create Table Operation
+
+Add a *Create Table* operation to create a new table based on a POST request that uses the template created in the first flow: (*Reviewers, please make this text more accurate. I am guessing here.*)
```
Not sure what that description is but, we need to add a create table operation in ordeer to create that table on the keyspace that was created before, create table operations takes as parameters a keyspace name, a table name, and a list columns that have a name, a type and a flag if they are primary key or not.  I hope I gave you enough info.

dhirsch1 committed:
Added text related to answers to questions in the landing page and examples topics

dhirsch1 committed:
Merge branch 'latest' into DOCS-9178-Cassandra-4.0-DH

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc:
```diff
@@ -0,0 +1,40 @@
+
+. After installing Apache Cassandra, open the `conf\cassandra.yaml` file for editing.
+. Optionally, change the value of `AllowAllAuthenticator` to `PasswordAuthenticator`.
+. Restart Cassandra.
+. In a separate terminal tab, verify that Cassandra is running:
```
```suggestion
. Restart Apache Cassandra.
. In a separate terminal tab, verify that Apache Cassandra is running:
```

kevintroller replied:
Just to make it consistent every time you call Cassandra.

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc:
```diff
@@ -0,0 +1,40 @@
+== POJO Configuration
+
+The connector exposes multiple operations in the Apache Cassandra Connector Reference.
+The below operations accept some POJOs as input parameters, which are needed for the DataSense functionality:
```
```suggestion
The following operations accept some POJOs as input parameters, which are needed for the DataSense functionality:
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc:
```diff
@@ -0,0 +1,40 @@
+The connector exposes multiple operations in the Apache Cassandra Connector Reference.
+The below operations accept some POJOs as input parameters, which are needed for the DataSense functionality:
+
+`addNewColumn`, `changeColumnType`, `createKeyspace`, `createTable`, `deleteColumnsValue`, `deleteRows`, `dropColumn`, `dropKeyspace`, `dropTable`, `executeCQLQuery`, `getTableNamesFromKeyspace`, `renameColumn`, `select`, and `update`.
```
```suggestion
* `addNewColumn` 
* `changeColumnType` 
* `createKeyspace` 
* `createTable` 
* `deleteColumnsValue` 
* `deleteRows` 
* `dropColumn` 
* `dropKeyspace` 
* `dropTable` 
* `executeCQLQuery` 
* `getTableNamesFromKeyspace` 
* `renameColumn` 
* `select` 
* `update`
```

kevintroller replied:
I think it's easier to scan, but feel free to update it or not.

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-design-center.adoc:
```diff
@@ -0,0 +1,77 @@
+. Select *+* to add a component.
+. Select the connector as a component.
+. Select the operation.
+. Click the Cassandra connector to configure the host, port, keyspace, and authentication:
```
```suggestion
. Click Cassandra Connector to configure the host, port, keyspace, and authentication:
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+The following examples show how to configure Apache Cassandra Connector:
+
+* <<create-table,Create a New Apache Cassandra Table>>
++
+Creates a template for an HTML form to use to enter data for the new table, creates a keyspace for the new table, and then creates the table
+* <<manipulate-data,Manipulate Data in an Apache Cassandra Database>>
```
```suggestion
* <<create-table,Create a New Apache Cassandra Table>> +
Creates a template for an HTML form to use to enter data for the new table, creates a keyspace for the new table, and then creates the table
* <<manipulate-data,Manipulate Data in an Apache Cassandra Database>>
Manipulate data in an Apache Casandra database by getting table names from a keyspace, performing CRUD operations, executing CQL queries and so on.
```

kevintroller replied:
Add an explanation sentence for the second example? Something like the above...

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+* The second flow creates the keyspace for the new table.
+* The third flow creates the new table.
+
+This example variables for some field values. You can either:
```
```suggestion
This example uses variables for some field values. You can either:
```

dhirsch1 committed:
Merge branch 'latest' into DOCS-9178-Cassandra-4.0-DH

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
```
```suggestion
. On the *Listener* properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+
+. From the *Mule Palette* view, select *CassandDB* and drag the *Create Table* operation to the right of the *Transform Message* component.
+. In the *Create Table* configuration, click the plus sign (*+*) next to the *Connector configuration* field.
+. Select global element that you configured for the *Create keyspace* operation.
```
```suggestion
. Select the global element that you configured for the *Create keyspace* operation.
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
++
+If the client test is successful, click **OK** to save the configuration. Otherwise, review and correct any invalid parameters and retest.
++
+. Run the Mule app.
```
```suggestion
. Save and run the Mule app.
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:
```diff
@@ -0,0 +1,575 @@
+** Drop a column.
+* Execute a Cassandra Query Language (CQL) query.
+* Drop data from the Apache Cassandra database table.
+
```
Here jumps directly to the XML sample of the application with the operations. I would add an intro line of this XML

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc:
```diff
@@ -0,0 +1,1262 @@
+
+Anypoint Connector for Cassandra (Cassandra Connector) implements the DataStax Java Driver for Apache Cassandra 3.1.
+
+Release Notes: xref:release-notes::connector/cassandra-connector-release-notes-mule-4.adoc[Cassandra Connector Release Notes]
```
```suggestion

```

kevintroller replied:
This is in the box of the Connectors Landing page.

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc:
```diff
@@ -0,0 +1,1262 @@
+| Configuration | String | The name of the configuration to use. | | x
+| Table a| String |  The name of the table to use for the operation |  | x
+| Keyspace Name a| String |  The keyspace that contains the table to use |  |
+| Column Name a| String |  The name of the column to removed |  #[payload] |
```
```suggestion
| Column Name a| String |  The name of the column to remove |  #[payload] |
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc:
```diff
@@ -0,0 +1,1262 @@
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Table a| String |  The name of the table in which the entity is updated |  | x
```
```suggestion
| Table a| String |  The name of the table in which to update the entity |  | x
```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc:
```diff
@@ -0,0 +1,135 @@
+
+To add and configure a connector in Studio:
+
+. Create a Mule project.
+. Add the connector to your Mule project.
+. Configure a source for the connector's flow.
+. Add a connector operation to the flow.
+. Configure a global element for the connector.
+. Configure the other connector fields.
```
```suggestion
. <<create-mule-project,Create a Mule project>>.
. <<add-connector-to-project,Add the connector to your Mule project>>.
. <<configure-input-source,Configure a source for the connector's flow>>.
. <<add-connector-operation,Add a connector operation to the flow>>.
. <<configure-global-element,Configure a global element for the connector>>.
. <<configure-other-fields,Configure the other connector fields>>.

```

kevintroller commented on cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc:
```diff
@@ -0,0 +1,135 @@
+[%header,cols="30s,70a"]
+|===
+|Field |Description
+|Endpoint Identification Algorithm | The endpoint identification algorithm used by clients to validate server host name.The default value is an empty string, which means it is disabled. Clients, including client connections created by the broker for inter-broker communication, verify that the broker host name matches the host name in the brokers certificate.
```
```suggestion
|Endpoint Identification Algorithm | The endpoint identification algorithm used by clients to validate server host name. The default value is an empty string, which means it is disabled. Clients, including client connections created by the broker for inter-broker communication, verify that the broker host name matches the host name in the brokers certificate.
```

dhirsch1 committed:
Added Kevin's edits

dhirsch1 committed:
Two fixes

dhirsch1 committed:
Merge branch 'latest' into DOCS-9178-Cassandra-4.0-DH

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-conf-task.adoc:
```diff
@@ -1,5 +1,4 @@
 = To Configure Cassandra Authentication
-:page-aliases: 3.9@mule-runtime::cassandra-connector-conf-task.adoc
```
These page aliases are specific to  Mule 3.9 and need to stay (page aliases that are specific to a particular version like in this case should not be removed).

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-keyspace-task.adoc:
```diff
@@ -1,5 +1,4 @@
 = To Create a Keyspace
-:page-aliases: 3.9@mule-runtime::cassandra-connector-create-keyspace-task.adoc
```
Don't delete

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-table-task.adoc:
```diff
@@ -1,5 +1,4 @@
 = To Create a Cassandra Table
-:page-aliases: 3.9@mule-runtime::cassandra-connector-create-table-task.adoc
```
Don't delete

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-datatypes.adoc:
```diff
@@ -1,5 +1,4 @@
 = Cassandra Connector Data Types Reference
-:page-aliases: 3.9@mule-runtime::cassandra-connector-datatypes.adoc
```
Don't delete

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-install-conf-task.adoc:
```diff
@@ -1,5 +1,4 @@
 = To Install and Configure the Cassandra Connector
-:page-aliases: 3.9@mule-runtime::cassandra-connector-install-conf-task.adoc
```
Don't delete

valkyrie69 commented on cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-ops-reference.adoc:
```diff
@@ -1,5 +1,4 @@
 = Cassandra Connector Operations Reference
-:page-aliases: 3.9@mule-runtime::cassandra-connector-ops-reference.adoc
```
Don't delete

valkyrie69 reviewed (changes requested):
Please reinstate the page aliases for the Mule 3.9 topics.

dhirsch1 committed:
Fixed Aliases in an earlier version, updated the driver version in the landing page topic

dhirsch1 committed:
Adding back Salesforce 10.5 text that I accidentally changed in this PR

dhirsch1 committed:
One change for an older alias

dhirsch1 committed:
Trying to fix a merge conflict
